src/coap_debug.c

```diff
@@ -77,19 +77,24 @@ static const char *loglevels[] = {
 COAP_STATIC_INLINE size_t
 print_timestamp(char *s, size_t len, coap_tick_t t) {
   struct tm *tmp;
+  int lensofar;
   time_t now = coap_ticks_to_rt(t);
   tmp = localtime(&now);
-  return strftime(s, len, "%b %d %H:%M:%S", tmp);
+  strftime(s, len, "%b %d %H:%M:%S", tmp);
+  lensofar = strlen(s);
+  snprintf(&s[lensofar], len-lensofar, ".%03u",
+           (unsigned int)((coap_ticks_to_rt_us(t) % 1000000)/1000));
+  return strlen(s);
 }
 
 #else /* alternative implementation: just print the timestamp */
 
 COAP_STATIC_INLINE size_t
 print_timestamp(char *s, size_t len, coap_tick_t t) {
 #ifdef HAVE_SNPRINTF
-  return snprintf(s, len, "%u.%03u",
+  return snprintf(s, len, "%u.%06u",
```

Comment:
As this PR now aims at milliseconds and we are printing `(coap_ticks_to_tr_us() % 1000000)/1000`, we should keep `%03u` here.

Reply:
Oops - corrected.